### PR TITLE
task 1 code changes

### PR DESCRIPTION
**Pull request by tejaswini-developer**

### CODE REVIEW

File: CODE_REVIEW.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+2. In Storage.service.ts, error handling is not implemented in readFromStorage() function while accessing an item in localstorage. Error should be caught and return meaning full message to user or initialize state accordingly. And also while setting item in local storage need to add error handling to catch storage full errors and display meaning full message to user
+3. Publisher name for 'Eloquent JavaScript' book is displayed empty in search list. This gives poor user experience empty cases to be handled to display explanatory messages like 'Publisher not mentioned'.
+4. Book description is displyed using innerHtml. Innerhtml usage could lead to Cross site scripting attacks and there is no proper validation of content provided, so broken html can be passed in which lead to security and unexpected problems.
+5. In total-count component unused ngOnInit() hook is defined that caused uneccesary implementation of OnInit interface. It is good practise to get rid of unused code causing uneccesary implementing of interface.
+6. In readinglist and totalcount components readingList$  and totalUnread$ are not declared as observables. Always declare type for variables in typescript
+
```
Comment (mandarini) [inline CODE_REVIEW.md:9]: You missed a few things, to be honest... Can you please check the code again and see if there are some subscriptions that are not getting unsubscribed, or they can be avoided by using some Angular built-in functionality? Same goes for some date formatting!

File: CODE_REVIEW.md [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+2. In Storage.service.ts, error handling is not implemented in readFromStorage() function while accessing an item in localstorage. Error should be caught and return meaning full message to user or initialize state accordingly. And also while setting item in local storage need to add error handling to catch storage full errors and display meaning full message to user
+3. Publisher name for 'Eloquent JavaScript' book is displayed empty in search list. This gives poor user experience empty cases to be handled to display explanatory messages like 'Publisher not mentioned'.
+4. Book description is displyed using innerHtml. Innerhtml usage could lead to Cross site scripting attacks and there is no proper validation of content provided, so broken html can be passed in which lead to security and unexpected problems.
+5. In total-count component unused ngOnInit() hook is defined that caused uneccesary implementation of OnInit interface. It is good practise to get rid of unused code causing uneccesary implementing of interface.
+6. In readinglist and totalcount components readingList$  and totalUnread$ are not declared as observables. Always declare type for variables in typescript
```
Comment (mandarini) [inline CODE_REVIEW.md:8]: This is not entirely true. Sometimes a type can be instantly inferred, so there is no need to specifically specify it, take a look at [this discussion](https://stackoverflow.com/questions/46799828/tslint-type-trivially-inferred-why-is-it-bad-practice-to-include-the-type-he) and [this documentation](https://palantir.github.io/tslint/rules/no-inferrable-types/)

**Review by mandarini — COMMENTED** [review]
Your solution does not pass linting, can you please fix the linting issues and address my comments? Also, why did you do this merge with the different branches? I assume it is because you had named your branch wrong. In this case, you can just checkout from your "wrong branch name" to a branch with the correct name, and push that. And then you should create a PR to your `master` branch, to follow the requirements of this task!

**Comment by tejaswini-developer** [comment]
> Your solution does not pass linting, can you please fix the linting issues and address my comments? Also, why did you do this merge with the different branches? I assume it is because you had named your branch wrong. In this case, you can just checkout from your "wrong branch name" to a branch with the correct name, and push that. And then you should create a PR to your `master` branch, to follow the requirements of this task!

Created pull request as per requirement now. And implemented changes as per review comments of this task in below pull request #5

File: CODE_REVIEW.md (hunk 1 above)
Comment (tejaswini-developer) [reply CODE_REVIEW.md:9]: solved in Pull Request #5

**Pull request merged**